### PR TITLE
Upgrade to JUnit 5

### PR DESCRIPTION
**Pull request by trev-dev**
Also removes some unused vars, imports and redundant class inheritance (while I was at it)

### CODE REVIEW

File: api/src/main/java/ca/bc/gov/educ/api/edx/struct/v1/EdxUserDistrictActivationRelinkSagaData.java [hunk 1]
```diff
@@ -5,12 +5,11 @@
 import lombok.EqualsAndHashCode;
 
 import jakarta.validation.constraints.NotNull;
-import java.io.Serializable;
 
 @EqualsAndHashCode(callSuper = true)
 @Data
 @JsonIgnoreProperties(ignoreUnknown = true)
-public class EdxUserDistrictActivationRelinkSagaData extends EdxUserDistrictActivationInviteSagaData implements Serializable {
```
Comment (mightycox) [inline api/src/main/java/ca/bc/gov/educ/api/edx/struct/v1/EdxUserDistrictActivationRelinkSagaData.java:13]: Is this something we should be removing? I thought we needed it on our structs

File: api/pom.xml [hunk 2]
```diff
@@ -186,23 +186,13 @@
       <artifactId>guava</artifactId>
       <version>${guava.version}</version>
     </dependency>
+    <!-- https://mvnrepository.com/artifact/org.junit.jupiter/junit-jupiter-engine -->
     <dependency>
       <groupId>org.junit.jupiter</groupId>
-      <artifactId>junit-jupiter-params</artifactId>
+      <artifactId>junit-jupiter-engine</artifactId>
```
Comment (mightycox) [inline api/pom.xml:192]: I think this dependency can be removed altogether

File: api/src/main/java/ca/bc/gov/educ/api/edx/orchestrator/CreateSchoolOrchestrator.java
Comment (mightycox) [inline api/src/main/java/ca/bc/gov/educ/api/edx/orchestrator/CreateSchoolOrchestrator.java:1]: Empty file can probably be removed

Comment (trev-dev) [reply api/src/main/java/ca/bc/gov/educ/api/edx/orchestrator/CreateSchoolOrchestrator.java:1]: I had removed this and re-added it in a squash, sweet! https://github.com/bcgov/EDUC-EDX-API/pull/214/commits/64c2e3722b5c1ed5c75825be56b92f550cf2600c

File: api/pom.xml (hunk 2 above)
Comment (trev-dev) [reply api/pom.xml]: ...looks like you're right :) https://github.com/bcgov/EDUC-EDX-API/pull/214/commits/bf76d207ba3e1abc1f0be7f66497e7b6c1ae293d

File: api/src/main/java/ca/bc/gov/educ/api/edx/struct/v1/EdxUserDistrictActivationRelinkSagaData.java (hunk 1 above)
Comment (trev-dev) [reply api/src/main/java/ca/bc/gov/educ/api/edx/struct/v1/EdxUserDistrictActivationRelinkSagaData.java:13]: Yep!  Serializable is already inherited by `EdxUserDistrictActivationInviteSagaData`, and so, is redundant here.

**Review by mightycox — APPROVED** [review]

**Pull request merged**